src/main/java/com/microsoft/graph/core/tasks/LargeFileUploadTask.java

```diff
@@ -275,9 +277,12 @@ private long nextSliceSize(long rangeBegin, long rangeEnd)  {
         long size = rangeEnd - rangeBegin + 1;
         return Math.min(size, this.maxSliceSize);
     }
-    private byte[] chunkInputStream(InputStream stream, int begin, int length) throws IOException {
-        byte[] buffer = new byte[length];
-        int lengthAssert = stream.read(buffer, begin, length);
+    private byte[] chunkInputStream(FileInputStream stream, int begin, int length) throws IOException {
+        ByteBuffer byteBuffer = ByteBuffer.allocate(length);
+        int lengthAssert = stream.getChannel().read(byteBuffer);
+        byteBuffer.flip();
+        byte[] buffer = byteBuffer.array();
+        byteBuffer.clear();
```

Comment:
I don't see how this code is meant to fix the [`IndexOutOfBoundsException`](https://docs.oracle.com/javase/8/docs/api/java/lang/IndexOutOfBoundsException.html). According to the Java [docs for the `read` method](https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read-byte:A-int-int-), this exception is thrown when:

> off is negative, len is negative, or len is greater than b.length - off

where `off` maps to `begin`, `len` maps to `length` and `b` maps to `buffer`.

Did I misunderstand the point of the PR?

Reply:
Hi @calebkiage .

According to the documentation:

> public int read(byte[] b, int off, int len) throws IOException
> 
> The first byte read is stored into element b[off], the next one into b[off+1], and so on.

However, from the second pass on, b[off] does not exist.